unity-renderer/Assets/DCLPlugins/BuilderInWorld/Scripts/BuiilderAPI/BuilderAPIController.cs

```diff
@@ -113,16 +115,22 @@ internal Promise<string> CallUrl(string method, string endpoint, string callPara
     }
 
     //This will disappear when we implement the signed fetch call
-    IEnumerator CallWeb (RequestHeader requestHeader, Promise<string> resultPromise, string contentType, bool isPut = true)
+    IEnumerator CallWeb (RequestHeader requestHeader, Promise<string> resultPromise, string contentType, string method)
     {
         UnityWebRequest www = null;
-        if (isPut)
-            www = UnityWebRequest.Put (BIWUrlUtils.GetBuilderAPIBaseUrl() + requestHeader.endpoint, requestHeader.body);
-        else
+        switch (method)
         {
-            WWWForm form = new WWWForm();
-            form.AddBinaryData("thumbnail", requestHeader.body);
-            www = UnityWebRequest.Post(BIWUrlUtils.GetBuilderAPIBaseUrl() + requestHeader.endpoint, form );
+            case PUT:
```

Comment:
Yes! Definitely! It should be implemented in our WebRequest handler, but since this code will disappear in the near future ( kernel will do this calls) I thought that It was going to be an overengineer to implement all the logic in WebRequest if it is going to be deleted